kdd-2024/ibis-tutorial-KDD2024.md

```diff
@@ -0,0 +1,240 @@
+
+**Expertise:**
+Chloe has presented talks and workshops across a number of conferences, such as DataConnect, NVIDIA GTC, and Google DevFest. She also taught lectures for the Machine Learning System Design course at Stanford and a crash course on recommendation systems.
+
+
+2. List of contributors, i.e., tutors who will only help prepare the tutorial material
+
+- Cody Peterson
+- Deepyaman Datta
+
+
+# Corresponding tutor with her/his email address
+
+- Naty Clementi 
+
+# Tutorial outline. Please provide as much detail as possible.
+
+
+## Intro and Setup “Going beyond pandas”
+
+Get attendees up and running in a GitHub Codespace or on their laptops. A bit of
+motivation about the kinds of problems where Ibis can help, and a general survey 
+of attendees to find out what their existing pain points and experiences are.
+
+## Introduction to Ibis basics
+
+**Slides**
+- What is Ibis?: Ibis as part of a composable data ecosystem.
+- Why Ibis?: present the problems that Ibis tackles and the benefits of adopting it.
+- Who Is Ibis for?: show how data engineers, data analysts, data scientists, and
+  any data practitioner can benefit from Ibis
+
+**Get familiarized with basics:**
+
+A hands-on, follow-along notebook introducing the basic verbs of Ibis data 
+analysis (select, filter, group_by, order_by, and aggregate), demonstrating a 
+simple example on a few local query engines with hands-on exercises throughout.
+
+## In-memory tables, joins, and data analysis
+
+Building on the previous notebook, we'll explore how to join in-memory data (from 
+a pandas DataFrame, polars DataFrame, Python dictionary, or PyArrow Table) with
+existing tables in  a local database and continue analysis on the join result.
+
+We'll explore chained joins, and demonstrate `read_parquet` and other `read_*` methods
+for loading local data into existing databases. Then we'll continue with a series 
+of hands-on exercises, building up an analysis pipeline for some IMDB ratings data,
+but only operating on a 5% sample of the original dataset. After, we show how the 
+same expression can be computed on the full dataset without any code changes, both 
+for local execution, or with bursting to a cloud database (or other hosted database).
+
+## Streaming 
+
+We'll build on our previous batch analysis example and turn what we've learned
+towards a similar problem, but now with a real-time and constantly updating data
+source. 
+
+With hands-on exercises, we'll explore how to set up rudimentary anomaly
+detection, and demonstrate how to window and group streaming data-sources before
+redirecting them to downstream sinks.
+
+## IbisML
```

Comment:
@lostmygithubaccount or @chloeh13q can you help writing one sentence of what we will aim to do here?

Reply:
"We will demonstrate how to define machine learning features, create data transformation pipelines, and exeucte them across backends using IbisML. We will then use these features to train an ML model!"

Reply:
I think the above is good, but we might want to mention something about how you can also do this without the usual memory constraints

Reply:
I added what Chloe mention in the docs, would you mind taking a crack at modifying the sentence to "mention something about how you can also do this without the usual memory constraints" 

I'm not fully sure, what we want to say here, by usual memory constraints . Is it like "running out of memory during and xgboost training" kind of thing?

Reply:
Yeah, I think this is less important to cover in the outline, so I would leave it as is.